SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -201,48 +229,77 @@ void USpatialNetDriver::InitiateConnectionToSpatialOS(const FURL& URL)
 
 	if (!bPersistSpatialConnection)
 	{
-		// Destroy the old connection
 		GameInstance->DestroySpatialWorkerConnection();
-
-		// Create a new SpatialWorkerConnection in the SpatialGameInstance.
 		GameInstance->CreateNewSpatialWorkerConnection();
 	}
 
 	Connection = GameInstance->GetSpatialWorkerConnection();
 
-	if (URL.HasOption(TEXT("locator")))
+	bool bShouldLoadFromURL = true;
+	bool bUseReceptionist = true;
+
+	// If this is the first connection try using the command line arguments to setup the config objects.
+	// If arguments can not be found we will use the regular flow of loading from the input URL.
+	if (!GameInstance->bFirstConnectionToSpatialOSAttempted || URL.Host == SpatialConstants::RECONNECT_USING_ARGUMENTS)
 	{
-		// Obtain PIT and LT.
-		Connection->LocatorConfig.PlayerIdentityToken = URL.GetOption(TEXT("playeridentity="), TEXT(""));
-		Connection->LocatorConfig.LoginToken = URL.GetOption(TEXT("login="), TEXT(""));
-		Connection->LocatorConfig.UseExternalIp = true;
-		Connection->LocatorConfig.WorkerType = GameInstance->GetSpatialWorkerType().ToString();
+		FString CommandLineLocatorHost;
+		FParse::Value(FCommandLine::Get(), TEXT("locatorHost"), CommandLineLocatorHost);
+		if(!CommandLineLocatorHost.IsEmpty())
+		{
+			bShouldLoadFromURL = !Connection->LocatorConfig.TryLoadCommandLineArgs();
+			bUseReceptionist = false;
+		}
+		else
+		{
+			bShouldLoadFromURL = !Connection->ReceptionistConfig.TryLoadCommandLineArgs();
+			bUseReceptionist = true;
+		}
+
+		GameInstance->bFirstConnectionToSpatialOSAttempted = true;
 	}
-	else // Using Receptionist
-	{
-		Connection->ReceptionistConfig.WorkerType = GameInstance->GetSpatialWorkerType().ToString();
 
-		// Check for overrides in the travel URL.
-		if (!URL.Host.IsEmpty() && URL.Host.Compare(SpatialConstants::LOCAL_HOST) != 0)
+	// Copy relevant URL info into the relevant config object.
+	if(bShouldLoadFromURL)
+	{
+		if (URL.Host == SpatialConstants::LOCATOR_HOST || URL.HasOption(TEXT("locator")))
 		{
-			Connection->ReceptionistConfig.ReceptionistHost = URL.Host;
+			FLocatorConfig& LocatorConfig = Connection->LocatorConfig;
```

Comment:
So this is still implicitly scanning the command line args for "+locatorHost"?

Reply:
Constructing a FLocatorConfig will not load it from the command line. FLocatorConfig has a function that can be called if you want to load it from the command line. This is only called in the section that checks:

(!GameInstance->GetFirstConnectionToSpatialOSAttempted() || URL.Host == SpatialConstants::RECONNECT_USING_COMMANDLINE_ARGUMENTS)